src/app/components/OptimoPromos/index.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useContext } from 'react';
+import { node, string, shape } from 'prop-types';
+import { ServiceContext } from '#contexts/ServiceContext';
+import path from 'ramda/src/path';
+
+import Timestamp from '#components/Promo/timestamp';
+import LiveLabel from '#legacy/psammead-live-label/src';
+import useViewTracker from '#hooks/useViewTracker';
+
+import Title from './Title';
+import Link from './Link';
+import MediaIndicator from './MediaIndicator';
+import Content from './Content';
+import { PromoWrapper, ContentWrapper } from './index.styles';
+import Image from './Image';
+import PromoContext from './PromoContext';
+
+const Promo = ({
+  children,
+  to,
+  ariaLabelledBy,
+  mediaType,
+  eventTrackingData,
+}) => {
+  const { service } = useContext(ServiceContext);
+
+  const eventTrackingDataSend = path(['block'], eventTrackingData);
+
+  const viewRef = useViewTracker(eventTrackingDataSend);
+
+  return (
+    <PromoWrapper ref={viewRef}>
+      <PromoContext.Provider
+        value={{ service, to, ariaLabelledBy, eventTrackingData, mediaType }}
+      >
+        {children}
+      </PromoContext.Provider>
+    </PromoWrapper>
+  );
+};
+
+Promo.ContentWrapper = ContentWrapper;
+Promo.Title = Title;
+Promo.MediaIndicator = MediaIndicator;
+Promo.Link = Link;
+Promo.Content = Content;
+
+// Outside OptimoPromos
+Promo.Timestamp = Timestamp;
+Promo.LiveLabel = LiveLabel;
+Promo.Image = Image;
+
+Promo.propTypes = {
+  children: node.isRequired,
+  to: string,
+  ariaLabelledBy: string.isRequired,
+  mediaType: string,
+  eventTrackingData: shape({ block: shape({ componentName: string }) }),
+};
+
+Promo.defaultProps = { to: '', mediaType: '', eventTrackingData: null };
```

Comment:
More default props, does the linting make you do this?

Reply:
Yes and No. It's my decision to not require those props as I think they are optional. It is the linter that requires me to have a default prop if I don't require the prop.